femmt/component.py

```diff
@@ -732,7 +744,161 @@ def excitation(self, frequency: float, amplitude_list: List, phase_deg_list: Lis
                     self.current[num] = complex(amplitude_list[num]*np.cos(np.deg2rad(phase_deg_list[num])),
                                                 amplitude_list[num]*np.sin(np.deg2rad(phase_deg_list[num])))
 
-        # Imposed current density
+
+
+                            # Imposed current density
+        if self.flag_excitation_type == 'current_density':
+            raise NotImplementedError
+
+        # Imposed voltage
+        if self.flag_excitation_type == 'voltage':
+            raise NotImplementedError
+
+
+        # -- Frequency --
+
+        self.frequency = frequency  # in Hz
+
+        # Define reduced frequency (used for homogenization technique)
+        # self.red_freq = np.empty(2)
+        self.red_freq = []
+        for num in range(len(self.windings)):
+            self.red_freq.append([])
+
+        if self.frequency != 0:
+            self.delta = np.sqrt(2 / (2 * self.frequency * np.pi * self.windings[0].cond_sigma * mu_0))  # TODO: distinguish between material conductivities
+            for num in range(len(self.windings)):
+                if self.windings[num].conductor_type == ConductorType.RoundLitz:
+                    self.red_freq[num] = self.windings[num].strand_radius / self.delta
+                elif self.windings[num].conductor_type == ConductorType.RoundSolid:
+                    self.red_freq[num] = self.windings[num].conductor_radius / self.delta
+                else:
+                    ff.femmt_print("Reduced Frequency does not have a physical value here")
+                    ff.femmt_print(self.windings[num].conductor_type)
+                    self.red_freq[num] = 1  # TODO: doesn't make sense like this -> rewrite fore conductor windings shape
+        else:
+            # DC case
+            self.delta = 1e20  # random huge value
+            self.red_freq[num] = 0
+
+    def excitation_time_domain(self, frequency: float, current_list: List[List[float]], time_list: List[float], time_period: float, initial_time: float, timemax : float, NbSteps: int, delta_time: float, phase_deg_list: List = None, ex_type: str = 'current',
+                   plot_interpolation: bool = False, imposed_red_f=0):
+        """
+         Excites the electromagnetic problem in the time domain with specified current and time settings.
+
+        Parameters:
+        - frequency (float): The frequency of the excitation in Hz.
```

Comment:
this text needs to go into the descriptions :param and :type
->
:param frequency: The frequency of the excitation in Hz
:type frequency: float